cmd/influx_tools/parquet/schema.go

```diff
@@ -0,0 +1,273 @@
+		case influxql.Boolean:
+			s.typeConverters[field] = toBool
+		case influxql.String:
+			s.typeConverters[field] = toString
+		default:
+			panic(fmt.Errorf("unknown converter %v for field %q", ftype, field))
+		}
+	}
+	return s
+}
+
+func (s *schemaCreator) extractSchema(ctx context.Context) error {
+	// Iterate over the shards and extract all series
+	for _, shard := range s.shards {
+		// Extract all fields of the measurement
+		fields := shard.MeasurementFields([]byte(s.measurement)).FieldSet()
+		if len(fields) == 0 {
+			continue
+		}
+
+		// Collect all available series in the shard and measurement and store
+		// them for later use and series accumulation
+		seriesCursor, err := shard.CreateSeriesCursor(
+			ctx,
+			tsdb.SeriesCursorRequest{},
+			influxql.MustParseExpr("_name = '"+s.measurement+"'"),
+		)
+		if err != nil {
+			return fmt.Errorf("getting series cursor failed: %w", err)
+		}
+		defer seriesCursor.Close()
+
+		for {
+			cur, err := seriesCursor.Next()
+			if err != nil {
+				return fmt.Errorf("advancing series cursor failed: %w", err)
+			}
+			if cur == nil {
+				break
+			}
+			mname := string(cur.Name)
+			if mname != s.measurement {
+				continue
+			}
+
+			s.series[shard.ID()] = append(s.series[shard.ID()], seriesEntry{
+				key:    s.measurement + "." + string(cur.Tags.HashKey(true)),
+				tags:   cur.Tags.Clone(),
+				fields: fields,
+			})
+		}
+	}
+
+	// Collect all tags and fields for creating the overall schema
+	tags := make(map[string]bool)
+	conflicts := make(map[string]map[influxql.DataType]bool)
+	s.fields = make(map[string]influxql.DataType)
+	for _, series := range s.series {
+		for _, serie := range series {
+			// Dedup tag keys
+			for _, tag := range serie.tags {
+				tags[string(tag.Key)] = true
+			}
+			// Detect field type conflicts and collect the fields
+			for name, current := range serie.fields {
+				if existing, found := s.fields[name]; found && current != existing {
+					if _, exists := conflicts[name]; !exists {
+						conflicts[name] = make(map[influxql.DataType]bool)
+					}
+					conflicts[name][current] = true
+					conflicts[name][existing] = true
+				}
+				s.fields[name] = current
+			}
+		}
+	}
+
+	// Apply the type resolutions
+	for name, ftype := range s.typeResolutions {
+		if _, found := s.fields[name]; !found {
+			continue
+		}
+		s.fields[name] = ftype
+	}
+
+	// Boil down all tags, fields and conflicts for later reuse
+	s.tags = make([]string, 0, len(tags))
+	for name := range tags {
+		s.tags = append(s.tags, name)
+	}
+	sort.Strings(s.tags)
+
+	s.fieldKeys = make([]string, 0, len(s.fields))
+	for name, ftype := range s.fields {
+		// Detect unconvertible fields
+		switch ftype {
+		case influxql.Float, influxql.Integer, influxql.String, influxql.Boolean, influxql.Unsigned:
+			// Accepted type
+			s.fieldKeys = append(s.fieldKeys, name)
+		default:
+			// Unconvertible type
+			panic(fmt.Errorf("unconvertible field %q with type %v", name, ftype))
```

Comment:
panic or better to return an error?  Or ignore that field and try to continue?

Reply:
Decided to return an error. I don't want to start out too permissive so let's error out here for now and decide for a more graceful error handling if need be.